pyne/mesh.py

```diff
@@ -1510,6 +1511,102 @@ def _do_op(self, other, tags, op, in_place=True):
 
         return mesh_1
 
+
+class MeshTally(StatMesh):
+    """This class stores all information from all single mesh tally that
+    exists within some meshtal or state point file. Header information is
+    stored as attributes and the "mesh" attribute is a MOAB mesh with all
+    result and relative error data tagged. This class inherits from StatMesh,
+    exposing all statistical mesh manipulation methods.
+    Attributes
+    ----------
+    tally_number : int
+        The tally number.
+        For mesh tally from MCNP, it must end with 4 (e.g. 4, 14, 214).
+        For mesh tally from OpenMC, it could be any int.
+    particle : string
+        Either "neutron" for a neutron mesh tally or "photon" for a photon mesh
+        tally.
+    dose_response : bool
+        True is the tally is modified by a dose response function.
+    x_bounds : list of floats
+        The locations of mesh vertices in the x direction.
+    y_bounds : list of floats
+        The locations of mesh vertices in the y direction.
+    z_bounds : list of floats
+        The locations of mesh vertices in the z direction.
+    dims : list
+        Dimensions of the mesh.
+    num_ves : int
+        Number of volume elements.
+    e_bounds : list of floats
+        The minimum and maximum bounds for energy bins.
+    num_e_groups: int
+        Number of energy groups.
+    mesh :
+        An PyMOAB core instance tagged with all results and
+        relative errors.
+    tag_names : iterable
+        Four strs that specify the tag names for the results, relative errors,
+        total results, and relative errors of the total results.
+    Notes
+    -----
+    All Mesh/StatMesh attributes are also present via a super() call to
+    StatMesh.__init__().
+    """
+
+    def __init__(self):
+        """Create an empty MeshTally object and set default values."""
+        if not HAVE_PYMOAB:
+            raise RuntimeError("PyMOAB is not available, "
```

Comment:
This should maybe raise `NotImplemented`? Not :100: on that though.

Reply:
Thsi `RuntimeError` is replaced with `NotImplementedError`. This link maybe useful for this type of error: https://help.semmle.com/wiki/display/PYTHON/NotImplemented+is+not+an+Exception.